src/core/Akka.Remote/Serialization/LruBoundedCache.cs

```diff
@@ -0,0 +1,110 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Akka.Remote.Serialization
+{
+    /// <summary>
+    /// INTERNAL API
+    /// </summary>
+    /// <remarks>
+    /// Fast hash based on the 128 bit Xorshift128+ PRNG. Mixes in character bits into the random generator state.
+    /// </remarks>
+    public static class FastHash
+    {
+        public static int OfString(string s)
+        {
+            var chars = s.ToCharArray();
+            var s0 = 391408L;
```

Comment:
I agree, that this sort of magic numbers should be described before merging to avoid confusion.